### PR TITLE
Money type

### PR DESCRIPTION
**Pull request by dgon-jd**
Added money attribute. 
Config: 
- receives list of currencies. If empty, uses default hardcoded list
- receives default currency id for this attribute

instance: 
- receives value

### CODE REVIEW

File: CloudFabric.EAV.Domain/Models/Attributes/Currency.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+namespace CloudFabric.EAV.Domain.Models.Attributes;
+
+public class Currency
```
Comment (ustims) [inline CloudFabric.EAV.Domain/Models/Attributes/Currency.cs:3]: https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/records

and 

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record#value-equality

I suppose that will remove the need for overriding Equals and other methods

**Review by ustims — CHANGES_REQUESTED** [review]

File: CloudFabric.EAV.Tests/Tests.cs [hunk 2]
```diff
@@ -994,6 +994,170 @@ public async Task CreateNumberAttribute_ValidationError()
...
+                new() { CultureInfoId = cultureInfoId, String = "test" }
+            },
+            Attributes = new List<EntityAttributeConfigurationCreateUpdateRequest> { moneyAttribute }
+        };
+
+        (EntityConfigurationViewModel? created, _) =
+            await _eavService.CreateEntityConfiguration(configCreateRequest, CancellationToken.None);
+        created.Attributes.Count.Should().Be(1);
+
+        ProjectionQueryResult<AttributeConfigurationListItemViewModel> allAttributes =
```
[the remark this answers is not in the document]
Comment (dgon-jd) [reply CloudFabric.EAV.Tests/Tests.cs:1031]: This is a copy-paste from many other similar tests. What is the problem specifically?

File: CloudFabric.EAV.Domain/Models/Attributes/Currency.cs (hunk 1 above)
Comment (dgon-jd) [reply CloudFabric.EAV.Domain/Models/Attributes/Currency.cs:3]: Make sense. Changed to a record.

File: CloudFabric.EAV.Domain/Models/Attributes/MoneyAttributeConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    public override List<string> ValidateInstance(AttributeInstance? instance)
+    {
+        var errors = base.ValidateInstance(instance);
+        if (instance is not MoneyAttributeInstance moneyInstance)
+        {
+            errors.Add("Cannot validate attribute. Expected attribute type: Money)");
+            return errors;
+        }
+
+        if (moneyInstance.Value < 0)
```
[the remark this answers is not in the document]
Comment (dgon-jd) [reply CloudFabric.EAV.Domain/Models/Attributes/MoneyAttributeConfiguration.cs]: Removed

**Review by ustims — APPROVED** [review]

File: CloudFabric.EAV.Domain/Models/Attributes/MoneyAttributeConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        List<LocalizedString>? description = null,
+        bool isRequired = false,
+        Guid? tenantId = null,
+        string? metadata = null) : base(id, machineName, name, EavAttributeType.Money, description, isRequired, tenantId, metadata)
+    {
+        currencies ??= DefaultListOfCurrencies();
+        Apply(new MoneyAttributeConfigurationUpdated(id, defaultCurrencyId, currencies));
+    }
+
+    private List<Currency> DefaultListOfCurrencies() => new List<Currency>
```
Comment (ustims) [inline CloudFabric.EAV.Domain/Models/Attributes/MoneyAttributeConfiguration.cs:32]: Where's bitcoin? 🪙

File: CloudFabric.EAV.Domain/Models/Attributes/MoneyAttributeConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        }
+        errors.Add("Cannot validate attribute. Expected attribute type: Money)");
+        return errors;
+    }
+
+    #endregion
+
+    #region Equality
+
+    public override bool Equals(object obj)
```
Comment (ustims) [inline CloudFabric.EAV.Domain/Models/Attributes/MoneyAttributeConfiguration.cs:69]: Do we still need those methods after switching to record?

File: CloudFabric.EAV.Tests/Tests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ustims) [reply CloudFabric.EAV.Tests/Tests.cs:1031]: maybe this runs on InMemory/Postgresql synchronous repositories. It's okay, I just noticed that there is no wait for projections, after creating an entity config it immediately queries them

**Pull request merged**